### PR TITLE
added factories back

### PR DESCRIPTION
**Pull request by raduchis**
## Reasoning behind the pull request
- split from refactor-factories-2
  
## Proposed changes
- more factories added

## Testing procedure
- all in, nothing should be broken

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: process/block/preprocess/sovereignSmartContractResultPreProcessorFactory.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package preprocess
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+// SmartContractResultPreProcessorCreator defines the interface of a smart contract result pre-processor creator
+type SmartContractResultPreProcessorCreator interface {
```
Comment (SebastianMarian) [inline process/block/preprocess/sovereignSmartContractResultPreProcessorFactory.go:9]: Could be this moved in process/interface.go or process/block/interface.go or process/block/preprocess/interface.go ?

File: process/factory/shard/preProcessorsContainerFactory.go [hunk 2]
```diff
@@ -177,33 +178,50 @@ func (ppcf *preProcessorsContainerFactory) createTxPreProcessor() (process.PrePr
 	return preprocess.NewTransactionPreprocessor(args)
 }
 
+// SmartContractResultPreProcessorCreator defines the interface of a smart contract result pre-processor creator
+type SmartContractResultPreProcessorCreator interface {
```
Comment (SebastianMarian) [inline process/factory/shard/preProcessorsContainerFactory.go:182]: Could be this moved in process/interface.go or process/factory/interface.go or process/factory/shard/interface.go ?

File: process/smartContract/processorV2/sovereignSCProcessFactory.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+func NewSovereignSCProcessFactory(creator scrCommon.SCProcessorCreator) (*sovereignSCProcessFactory, error) {
+	if check.IfNil(creator) {
+		return nil, process.ErrNilSCProcessorCreator
+	}
+	return &sovereignSCProcessFactory{
+		scrProcessorCreator: creator,
+	}, nil
+}
+
+// CreateSCProcessor CreateSCRProcessor creates a new smart contract processor
```
Comment (SebastianMarian) [inline process/smartContract/processorV2/sovereignSCProcessFactory.go:23]: Delete CreateSCRProcessor

File: process/block/preprocess/smartContractResultPreProcessorCreatorArgs.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/state"
+)
+
+type SmartContractResultPreProcessorCreatorArgs struct {
```
[the remark this answers is not in the document]
Comment (raduchis) [reply process/block/preprocess/smartContractResultPreProcessorCreatorArgs.go:14]: fixed

File: process/block/preprocess/sovereignSmartContractResultPreProcessorFactory.go (hunk 1 above)
Comment (raduchis) [reply process/block/preprocess/sovereignSmartContractResultPreProcessorFactory.go:9]: moved to process/block/preprocess/interface.go

File: process/smartContract/processorV2/sovereignSCProcessFactory.go (hunk 3 above)
Comment (raduchis) [reply process/smartContract/processorV2/sovereignSCProcessFactory.go:23]: fixed

File: process/factory/shard/preProcessorsContainerFactory.go (hunk 2 above)
Comment (raduchis) [reply process/factory/shard/preProcessorsContainerFactory.go]: moved to process/block/preprocess/interface.go to not have cyclic dependency

**Review by SebastianMarian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**